tiledb/sm/query/readers/reader_base.cc

```diff
@@ -1718,6 +1718,12 @@ Status ReaderBase::unfilter_tile_nullable(
   return Status::Ok();
 }
 
+uint64_t ReaderBase::offsets_bytesize() const {
+  assert(offsets_bitsize_ == 32 || offsets_bitsize_ == 64);
```

Comment:
This invariant should ideally be enforced in the constructor (it is already enforced in the setter in strategy_base).

Reply:
It's actually enforced in init_read_state as well, which should be moved to the constructor when we do C.41 compliance for the readers. I'll just remove the assert here as it's not necessary.